pkg/controllers/workgenerator/controller_integration_test.go

```diff
@@ -426,33 +429,67 @@ var _ = Describe("Test Work Generator Controller", func() {
-					// mark the work as not applied with failed manifests
-					markWorkWithFailedToApplyAndNotAvailable(&work)
+					// mark the work as not applied with failed manifests, and diffed manifests
+					markWorkWithFailedToApplyAndNotAvailable(&work, true, true)
 					// check the binding status that it should have failed placement
-					verifyBindStatusNotAppliedWithFailedPlacement(binding, false) // mark the work applied but still not available with failed manifests
+					verifyBindStatusNotAppliedWithTwoPlacements(binding, false, true, true, true) // mark the work applied but still not available with failed manifests
 					// mark the work available directly
 					markWorkAvailable(&work)
-					// check the binding status that it should be marked as available true eventually
-					verifyBindStatusAvail(binding, false)
+					// check the binding status that it should be marked as available true eventually with 2 drift placements
+					verifyBindStatusAvail(binding, false, true)
 				})
 
 				It("Should update the binding when the failed placement list has changed but over all condition didn't change", func() {
 					// mark the work as not applied with failed manifests
-					markWorkWithFailedToApplyAndNotAvailable(&work)
+					markWorkWithFailedToApplyAndNotAvailable(&work, false, false)
 					// check the binding status that it should have failed placement
-					verifyBindStatusNotAppliedWithFailedPlacement(binding, false) // mark the work applied but still not available with failed manifests
+					verifyBindStatusNotAppliedWithTwoPlacements(binding, false, true, false, false) // mark the work applied but still not available with failed manifests
 					// change one of the failed manifests to be applied
-					markWorkAsAppliedButNotAvailableWithFailedManifest(&work)
+					markWorkAsAppliedButNotAvailableWithFailedManifest(&work, false)
 					// check the binding status that it should be applied but not available and with two failed placement
-					verifyBindStatusNotAvailableWithTwoFailedPlacement(binding, false)
+					verifyBindStatusNotAvailableWithTwoPlacements(binding, false, true, false)
 					// mark one of the failed manifests as available but no change in the overall condition
-					markOneManifestAvailable(&work)
+					markOneManifestAvailable(&work, false)
 					// check the binding status that it should be applied but not available and with one failed placement
-					verifyBindStatusNotAvailableWithOneFailedPlacement(binding, false)
+					verifyBindStatusNotAvailableWithOnePlacement(binding, false, true)
 					// mark the work available directly
 					markWorkAvailable(&work)
 					// check the binding status that it should be marked as available true eventually
-					verifyBindStatusAvail(binding, false)
+					verifyBindStatusAvail(binding, false, false)
+				})
+
+				It("Should update the binding when the diffed placement list has changed but over all condition didn't change", func() {
```

Comment:
I wonder which step tested the "diffed placement list"?

Reply:
Anytime we verify the binding status. We make sure the binding status has the necessary and same diffed placement list as the work. In those functions, `hasDiffedPlacements` is a Boolean to add the diffed placements to the want status. Then we compare the binding status with the wanted binding status. The vice versa is the work no longer has diffed placements.

Reply:
The actual step is when `markWorkAsAppliedButNotAvailableWithFailedManifest` once Applied condition is true the diffed placement list should be nil.